packages/web-api/src/methods.ts

```diff
@@ -606,6 +606,13 @@ export interface OAuthAccessArguments extends WebAPICallOptions {
   client_secret: string;
   code: string;
   redirect_uri?: string;
+  single_channel?: string;
+}
+export interface OAuthV2AccessArguments extends WebAPICallOptions {
+  client_id: string;
```

Comment:
https://api.slack.com/methods/oauth.v2.access notes that `client_id` and `client_secret` are optional?

Reply:
i can try to verify, but i think they should be required.

Reply:
@shanedewael hmm, I believe they're required. Rather than that, I noticed that this package should send them using basic auth instead as [the api doc strongly recommends](https://api.slack.com/methods/oauth.v2.access) (it's the same for v1). I will create a GitHub issue for the basic-auth issue and then make the change in this PR.

Reply:
I've modified the implementation to send a base64 value in Authorization header instead of sending `client_id` and `client_secret` in `request.body` when detecting those in `options`. I believe this PR is ready to go.